includes/orders/functions/orders.php

```diff
@@ -1375,42 +1375,125 @@ function edd_get_net_order_statuses() {
 }
 
 /**
- * Get the order status array keys which are considered recoverable.
+ * Generate unique payment key for orders.
  *
  * @since 3.0
- * @return array An array of order status keys which are considered recoverable.
+ * @param string $key Additional string used to help randomize key.
+ * @return string
  */
-function edd_recoverable_order_statuses() {
-	$statuses = array( 'pending', 'abandoned', 'failed' );
+function edd_generate_order_payment_key( $key ) {
+	$auth_key    = defined( 'AUTH_KEY' ) ? AUTH_KEY : '';
+	$payment_key = strtolower( md5( $key . gmdate( 'Y-m-d H:i:s' ) . $auth_key . uniqid( 'edd', true ) ) );
 
 	/**
-	 * Order statuses which are considered recoverable.
+	 * Filters the payment key
+	 *
+	 * @since 3.0
+	 * @param string $payment_key The value to be filtered
+	 * @param string $key Additional string used to help randomize key.
+	 * @return string
+	 */
+	return apply_filters( 'edd_generate_order_payment_key', $payment_key, $key );
+}
+
+/**
+ * Helper functions to retrieve or statues for different phases of the life of an order.
+ */
+
+/**
+ * Get the order status array keys which are considered completed.
+ *
+ * @since 3.0
+ * @return array An array of order status keys which are considered completed.
+ */
+function edd_complete_order_status_keys() {
+	$statuses = array( 'publish', 'complete', 'completed', 'partially_refunded', 'revoked', 'refunded' );
+
+	/**
+	 * Order statuses which are considered completed or at their final state.
 	 *
 	 * @param $statuses {
 	 *        An array of order status array keys.
 	 * }
 	 */
-	return apply_filters( 'edd_recoverable_payment_statuses', $statuses );
+	return apply_filters( 'edd_complete_order_status_keys', $statuses );
 }
 
 /**
- * Generate unique payment key for orders.
+ * Get the order statuses which are considered completed.
  *
  * @since 3.0
- * @param string $key Additional string used to help randomize key.
- * @return string
+ * @return array An array of order statuses which are considered completed.
  */
-function edd_generate_order_payment_key( $key ) {
-	$auth_key    = defined( 'AUTH_KEY' ) ? AUTH_KEY : '';
-	$payment_key = strtolower( md5( $key . gmdate( 'Y-m-d H:i:s' ) . $auth_key . uniqid( 'edd', true ) ) );
+function edd_complete_order_statuses() {
+	return array_intersect_key( edd_get_payment_statuses(), array_flip( edd_complete_order_status_keys() ) );
+}
+
+/**
+ * Get the order status array keys which are considered incomplete.
+ *
+ * @since 3.0
+ * @return array An array of order status keys which are considered incomplete.
+ */
+function edd_incomplete_order_status_keys() {
```

Comment:
Two comments on this (and the new functions in general):
- I would prefer for them to be named `edd_get_` just because that makes it really clear that it is intended to return something. That would also make it consistent with the net/gross order status functions we already introduced (`edd_get_net_order_statuses`, eg).
- In that same vein, I would love for `edd_get_incomplete_order_statuses` to be consistent with the net/gross status functions, and return the same kind of array (sans labels).
- Do we _need_ the functions which return the statuses with labels? I feel like that introduces some extra complexity which may not be necessary.